lib/core/security/profileRepository.ts

```diff
@@ -436,9 +477,12 @@ class ProfileRepository extends Repository {
       throw kerror.get('security', 'profile', 'missing_anonymous_role');
     }
 
-    await this.persistToDatabase(profile, { method, refresh, retryOnConflict });
+    delete profile.optimizedPolicies; // Remove optimized policies
+    await super.persistToDatabase(profile, { method, refresh, retryOnConflict });
 
     const updatedProfile = await this.loadOneFromDatabase(profile._id);
+    // Recompute optimized policies based on new policies
+    updatedProfile.optimizedPolicies = this._optimizePolicies(updatedProfile.policies);
```

Comment:
The method `validateAndSaveProfile` is used when a profile has been updated to save it to the database,
the reason of why I'm deleting optimizedPolicies before saving to database is because I do not want to save the optimizedPolicies in the database, and the reason behind that is that since the method `validateAndSaveProfile` is used to update a profile by saving it to the database and retrieving the updated profile from the database to update it inside the profile map.
When this method is called we need to recompute the optimizedPolicies after they are loaded from the database because the array `policies` might have changed impliying that optimizedPolicies doesn't reflect the correct policies anymore.